src/handlers/s3.ts

```diff
@@ -68,7 +68,9 @@ export default function s3HandlerFactory({
       ? utils.resolveParams(`${resolvedEndpoint}/{file}`, ctx.params)
       : resolvedEndpoint; // Bucket operations
 
-    const headers = ctx.request.headers || {};
+    const headers = {
+      ...(ctx.request?.headers?.range && { range: ctx.request.headers.range }),
```

Comment:
Do we not want the etags to be forwarded?

Reply:
the current HEAD just passes on the range headers:-
https://github.com/markusahlstrand/cloudworker-proxy/blob/10c4ffc085c77a499bbe74bb5940ce4e8aa770e4/src/handlers/s3.ts#L74

I changed it to pass on everything in this PR earlier. Then I realised that broke the AWS signing so I reverted back to jsut the range headers. Will need to fix at some point but I am not sure which headers break the signer yet. An aws4fetch-like lib suggests the host header is a problem https://github.com/jamesmbourne/aws4-axios/issues/110

Reply:
the current HEAD initializes the headers to {} and then adds a range header only. We are indeed missing useful features like If-None-Match

Reply:
I could add if-none-match and other headers but I am unsure of whether they are capitalised or not